### PR TITLE
Add tests for backend/openshiftcluster

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Part of https://github.com/Azure/ARO-RP/issues/59 , as well as increasing test coverage

### What this PR does / why we need it:

Adds test coverage for backend/openshiftcluster and how it calls the Manager. This includes tests for (some) dequeuing as well.

### Test plan for issue:

This is tests :D

I would like to add some more tests for the fake database layer, though, to ensure that it's doing what we actually expect. Probably in a way that I can also run the same tests against the simulator to verify it. The goal is for the database clients to be verified-enough fakes.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: hack/gentestdb/gentestdb.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"flag"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+func main() {
```
Comment (asalkeld) [inline hack/gentestdb/gentestdb.go:15]: extra space

File: hack/gentestdb/gentestdb.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+type db struct {
+	Name              string
+	DocumentInterface string
+	UsesPartitions    bool
+	UsesLeases        bool
+}
+
+func Generate(pathName string) error {
```
Comment (asalkeld) [inline hack/gentestdb/gentestdb.go:44]: kill the new line at the beginning of functions

File: hack/gentestdb/gentestdb.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		}
+
+		err = tmpl.Execute(writer, f)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	return nil
```
Comment (asalkeld) [inline hack/gentestdb/gentestdb.go:94]: lots of extra new lines in this file

**Comment by ehashman** [comment]
@hawkowl can you please rebase this?

File: hack/gentestdb/gentestdb.go (hunk 1 above)
Comment (mjudeikis) [reply hack/gentestdb/gentestdb.go:15]: wrap this in 
```
func main(){
err:= run()
if err!=nil{
   panic(err)
}}
```
So we can have `return err` as a pattern like anywhere else in the code

**Review by mjudeikis — COMMENTED** [review]
Any reason this can;t be upstreamed into our cosmosdb package? Was this a consideration or nogo?

**Comment by hawkowl** [comment]
Will do this in another PR, with the relevant bits rebased....

**Pull request closed without merging**